src/feat/pitch-functions.cc

```diff
@@ -778,8 +778,9 @@ int32 OnlinePitchFeatureImpl::NumFramesAvailable(
     if (input_finished_ && !snip_edges) {
       return (int32)(num_downsampled_samples * 1.0f / frame_shift + 0.5f);
     }
-    else
-      return ((num_downsampled_samples - frame_length) / frame_shift) + 1;
+  // BP: handles snip_edges == false in a more graceful way
+    else if (!snip_edges) return (int32)((num_downsampled_samples - frame_length/2) * 1.0f / frame_shift + 0.5f);
```

Comment:
past the 80-char line length limit.
probably best to remove this comment.
you may need to use braces on the if-statement to stay within the line length limit.